### PR TITLE
feat: add wallet support

### PR DESCRIPTION
**Pull request by johnletey**

### CODE REVIEW

File: src/contract-create.ts [hunk 1]
```diff
@@ -13,11 +13,11 @@ import { JWKInterface } from 'arweave/node/lib/wallet';
  */
 export async function simulateCreateContractFromSource(
   arweave: Arweave,
-  wallet: JWKInterface,
+  wallet: JWKInterface | 'use_wallet',
   initState: string,
   contractSrc: string,
 ): Promise<Transaction> {
-  const srcTx = await arweave.createTransaction({ data: contractSrc }, wallet);
+  const srcTx = await arweave.createTransaction({ data: contractSrc });
```
[the remark this answers is not in the document]
Comment (johnletey) [reply src/contract-create.ts:20]: @samcamwilliams Yeah `createTransaction` already supports `use_wallet` 🙂 Can revert this!

**Comment by johnletey** [comment]
> but curious if this is actually necessary?

@samcamwilliams Yup, this is very important for sites interacting with SmartWeave, but don't want to ask the user for their JWK 🙃

**Review by luca-arweave — COMMENTED** [review]
I think that @samcamwilliams is right. The changes is actually not necessary because, actually, arweave.js should take care of understanding if users is using a JWK or the wallet. 

Anyway, to better understand, I need to test the AreweaveWallet. is there any beta that I can check?

**Comment by johnletey** [comment]
Okay @samcamwilliams @luca-arweave, reverted the change to `createTransaction`. Much simpler 🙃

In the meantime, I've shared the wallet source with Luca for him to test this with!

**Comment by luca-arweave** [comment]
amazing @johnletey ! Thanks! 


Can we close the PR, so?

**Comment by johnletey** [comment]
@luca-arweave Yup! Should be good to go! 🙌🏻

**Comment by luca-arweave** [comment]
👍 waiting for @cedriking to merge the PR!

**Pull request merged**